app/models/queue_item.rb

```diff
@@ -1,10 +1,38 @@
 class QueueItem < ActiveRecord::Base
+  validate :position_cannot_be_less_than_one
+  validate :position_cannot_be_greater_than_queue_items_count
+
   belongs_to :user
   belongs_to :video
 
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  def self.save_and_update_positions(queue_ids_with_positions)
+    begin
+      ActiveRecord::Base.transaction do
+        queue_ids_with_positions.each do |id, position|
+            queue_item = QueueItem.find(id.to_i)
+            queue_item.update!(position: position)
+        end
+      end
+    rescue ActiveRecord::RecordInvalid => e
+      return e.message
+    end
+  end
+
+  def position_cannot_be_less_than_one
+    if !position.nil? && position < 1
```

Comment:
`!position.nil?` is not very Ruby-ish. not nil - means either false or truthy `position`. I assume `false` can be skipped as we want truthy `position` - what you should learn is to use just `if position` or `if position.present?` - the latter is available in Rails.

Also in Ruby 2.3.0 you can use `position&.method` in case position would be nil the right part of the method is not executed.